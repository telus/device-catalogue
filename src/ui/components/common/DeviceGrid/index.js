import React from "react";
import PropTypes from "prop-types";
import Box from "@tds/core-box";
import Text from "@tds/core-text";
import FlexGrid from "@tds/core-flex-grid";
import phoneImage from "../../../../../public/images/iphone.jpg";
import watchImage from "../../../../../public/images/watch.jpg";
import DeviceCard from "../DeviceCard";

const DeviceGrid = ({ devices, deviceType }) => {
  if (!devices) {
    return (
      <Box inset={4}>
        <FlexGrid>
          <FlexGrid.Row>
            <Text>No devices found</Text>
          </FlexGrid.Row>
        </FlexGrid>
      </Box>
    );
  }
  const deviceImage = deviceType === "iphone" ? phoneImage : watchImage;

  return (
    <Box inset={0}>
      <FlexGrid>
        <FlexGrid.Row horizontalAlign="center">
          {devices.map((device, index) => {
            return (
              <FlexGrid.Col key={`device-${index}`} xs={12} md={3}>
                <DeviceCard
                  device={device}
                  image={deviceImage}
                  imageAlt="Image of an IPhone"
                />
              </FlexGrid.Col>
            );
          })}
        </FlexGrid.Row>
      </FlexGrid>
    </Box>
  );
};

DeviceGrid.propTypes = {
  devices: PropTypes.arrayOf(PropTypes.object),
  deviceType: PropTypes.string.isRequired,
};

export default DeviceGrid;
