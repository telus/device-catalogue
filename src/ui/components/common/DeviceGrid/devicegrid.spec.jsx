import React from "react";
import { render } from "enzyme";
import DeviceGrid from ".";

const iphones = [
  {
    name: "iPhone 12 Pro",
    brand: "Apple",
    price: 1399,
  },
  {
    name: "iPhone 12",
    brand: "Apple",
    price: 979,
  },
  {
    name: "iPhone SE",
    brand: "Apple",
    price: 599,
  },
];

const watches = [
  {
    name: "Apple Watch Series 6",
    brand: "Apple",
    price: 529,
  },
  {
    name: "Apple Watch SE",
    brand: "Apple",
    price: 369,
  },
  {
    name: "Apple Watch Series 3",
    brand: "Apple",
    price: 259,
  },
];

describe("Device Grid", () => {
  it("renders correctly with props for iphone devices", () => {
    const container = render(
      <DeviceGrid devices={iphones} deviceType="iphone" />
    );

    expect(container.html()).toMatchSnapshot();
    expect(container.find("img").attr("src").includes("iphone.jpg")).toEqual(
      true
    );
    expect(container.html().includes("iPhone 12")).toEqual(true);
  });

  it("renders correctly with props for watch devices", () => {
    const container = render(
      <DeviceGrid devices={watches} deviceType="watch" />
    );

    expect(container.html()).toMatchSnapshot();
    expect(container.find("img").attr("src").includes("watch.jpg")).toEqual(
      true
    );
    expect(container.html().includes("Apple Watch Series 6")).toEqual(true);
  });
});
