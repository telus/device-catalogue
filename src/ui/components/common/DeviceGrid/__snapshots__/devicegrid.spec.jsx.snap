// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device Grid renders correctly with props for iphone devices 1`] = `"<div class=\\"FlexGrid__StyledGrid-sc-1da2tb5-0 tOBOA\\"><div class=\\"Row__StyledRow-sc-1goz0ht-0 eqQOmX\\"><div class=\\"Col__StyledCol-sc-15yvjc7-0 cFUOhb\\"><div class=\\"sc-bdvvtL iRVQej\\"><div class=\\"sc-bdvvtL bzuMLt sc-hKwDye klxlDh sc-hKwDye klxlDh\\"><div class=\\"sc-bdvvtL iEjIjW\\"><div class=\\"sc-bdvvtL eCDRPO\\"><img src=\\"../public/images/iphone.jpg\\" width=\\"200\\" height=\\"300\\" alt=\\"Image of an IPhone\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"></div><div class=\\"sc-bdvvtL ipXlaM sc-pVTFL eoipii sc-pVTFL eoipii\\"><span data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 kwSJnX\\">Apple</span><span class=\\"sc-jrQzAO bBBXUK\\">iPhone 12 Pro</span></div><div class=\\"sc-bdvvtL iEjIjW sc-jRQBWg iapZSH sc-jRQBWg iapZSH\\"><div class=\\"sc-bdvvtL ipXlaM\\"><div class=\\"PriceLockup__StyledRateTextWrapper-crmhv4-4 inieGs\\"><div class=\\"sc-bdvvtL chLtDX sc-gKclnd gQsEUy sc-gKclnd gQsEUy\\"><div class=\\"sc-bdvvtL chLtDX\\"><span aria-hidden=\\"false\\" data-testid=\\"dollarSign\\" class=\\"PriceLockup__StyledDollarSign-crmhv4-2 bsmnqF\\">$</span><span data-testid=\\"priceValue\\" aria-hidden=\\"false\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\">1399</span><span data-testid=\\"priceValue\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\"><span class=\\"Text__StyledText-sc-1m0rr67-0 sc-iCfMLu fMvSxW fqtOUC\\"></span></span></div></div></div></div></div></div><div class=\\"sc-bdvvtL iaciPl\\"><hr class=\\"HairlineDivider__StyledHairlineDivider-biamlk-0 BvZCb\\"><p class=\\"Paragraph__StyledParagraph-efl81j-0 jukEfT\\">TELUS Easy Payment and Bring-It-Back applied. Read legal footnote Taxes due upfront.</p></div></div></div></div><div class=\\"Col__StyledCol-sc-15yvjc7-0 cFUOhb\\"><div class=\\"sc-bdvvtL iRVQej\\"><div class=\\"sc-bdvvtL bzuMLt sc-hKwDye klxlDh sc-hKwDye klxlDh\\"><div class=\\"sc-bdvvtL iEjIjW\\"><div class=\\"sc-bdvvtL eCDRPO\\"><img src=\\"../public/images/iphone.jpg\\" width=\\"200\\" height=\\"300\\" alt=\\"Image of an IPhone\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"></div><div class=\\"sc-bdvvtL ipXlaM sc-pVTFL eoipii sc-pVTFL eoipii\\"><span data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 kwSJnX\\">Apple</span><span class=\\"sc-jrQzAO bBBXUK\\">iPhone 12</span></div><div class=\\"sc-bdvvtL iEjIjW sc-jRQBWg iapZSH sc-jRQBWg iapZSH\\"><div class=\\"sc-bdvvtL ipXlaM\\"><div class=\\"PriceLockup__StyledRateTextWrapper-crmhv4-4 inieGs\\"><div class=\\"sc-bdvvtL chLtDX sc-gKclnd gQsEUy sc-gKclnd gQsEUy\\"><div class=\\"sc-bdvvtL chLtDX\\"><span aria-hidden=\\"false\\" data-testid=\\"dollarSign\\" class=\\"PriceLockup__StyledDollarSign-crmhv4-2 bsmnqF\\">$</span><span data-testid=\\"priceValue\\" aria-hidden=\\"false\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\">979</span><span data-testid=\\"priceValue\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\"><span class=\\"Text__StyledText-sc-1m0rr67-0 sc-iCfMLu fMvSxW fqtOUC\\"></span></span></div></div></div></div></div></div><div class=\\"sc-bdvvtL iaciPl\\"><hr class=\\"HairlineDivider__StyledHairlineDivider-biamlk-0 BvZCb\\"><p class=\\"Paragraph__StyledParagraph-efl81j-0 jukEfT\\">TELUS Easy Payment and Bring-It-Back applied. Read legal footnote Taxes due upfront.</p></div></div></div></div><div class=\\"Col__StyledCol-sc-15yvjc7-0 cFUOhb\\"><div class=\\"sc-bdvvtL iRVQej\\"><div class=\\"sc-bdvvtL bzuMLt sc-hKwDye klxlDh sc-hKwDye klxlDh\\"><div class=\\"sc-bdvvtL iEjIjW\\"><div class=\\"sc-bdvvtL eCDRPO\\"><img src=\\"../public/images/iphone.jpg\\" width=\\"200\\" height=\\"300\\" alt=\\"Image of an IPhone\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"></div><div class=\\"sc-bdvvtL ipXlaM sc-pVTFL eoipii sc-pVTFL eoipii\\"><span data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 kwSJnX\\">Apple</span><span class=\\"sc-jrQzAO bBBXUK\\">iPhone SE</span></div><div class=\\"sc-bdvvtL iEjIjW sc-jRQBWg iapZSH sc-jRQBWg iapZSH\\"><div class=\\"sc-bdvvtL ipXlaM\\"><div class=\\"PriceLockup__StyledRateTextWrapper-crmhv4-4 inieGs\\"><div class=\\"sc-bdvvtL chLtDX sc-gKclnd gQsEUy sc-gKclnd gQsEUy\\"><div class=\\"sc-bdvvtL chLtDX\\"><span aria-hidden=\\"false\\" data-testid=\\"dollarSign\\" class=\\"PriceLockup__StyledDollarSign-crmhv4-2 bsmnqF\\">$</span><span data-testid=\\"priceValue\\" aria-hidden=\\"false\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\">599</span><span data-testid=\\"priceValue\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\"><span class=\\"Text__StyledText-sc-1m0rr67-0 sc-iCfMLu fMvSxW fqtOUC\\"></span></span></div></div></div></div></div></div><div class=\\"sc-bdvvtL iaciPl\\"><hr class=\\"HairlineDivider__StyledHairlineDivider-biamlk-0 BvZCb\\"><p class=\\"Paragraph__StyledParagraph-efl81j-0 jukEfT\\">TELUS Easy Payment and Bring-It-Back applied. Read legal footnote Taxes due upfront.</p></div></div></div></div></div></div>"`;

exports[`Device Grid renders correctly with props for watch devices 1`] = `"<div class=\\"FlexGrid__StyledGrid-sc-1da2tb5-0 tOBOA\\"><div class=\\"Row__StyledRow-sc-1goz0ht-0 eqQOmX\\"><div class=\\"Col__StyledCol-sc-15yvjc7-0 cFUOhb\\"><div class=\\"sc-bdvvtL iRVQej\\"><div class=\\"sc-bdvvtL bzuMLt sc-hKwDye klxlDh sc-hKwDye klxlDh\\"><div class=\\"sc-bdvvtL iEjIjW\\"><div class=\\"sc-bdvvtL eCDRPO\\"><img src=\\"../public/images/watch.jpg\\" width=\\"200\\" height=\\"300\\" alt=\\"Image of an IPhone\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"></div><div class=\\"sc-bdvvtL ipXlaM sc-pVTFL eoipii sc-pVTFL eoipii\\"><span data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 kwSJnX\\">Apple</span><span class=\\"sc-jrQzAO bBBXUK\\">Apple Watch Series 6</span></div><div class=\\"sc-bdvvtL iEjIjW sc-jRQBWg iapZSH sc-jRQBWg iapZSH\\"><div class=\\"sc-bdvvtL ipXlaM\\"><div class=\\"PriceLockup__StyledRateTextWrapper-crmhv4-4 inieGs\\"><div class=\\"sc-bdvvtL chLtDX sc-gKclnd gQsEUy sc-gKclnd gQsEUy\\"><div class=\\"sc-bdvvtL chLtDX\\"><span aria-hidden=\\"false\\" data-testid=\\"dollarSign\\" class=\\"PriceLockup__StyledDollarSign-crmhv4-2 bsmnqF\\">$</span><span data-testid=\\"priceValue\\" aria-hidden=\\"false\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\">529</span><span data-testid=\\"priceValue\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\"><span class=\\"Text__StyledText-sc-1m0rr67-0 sc-iCfMLu fMvSxW fqtOUC\\"></span></span></div></div></div></div></div></div><div class=\\"sc-bdvvtL iaciPl\\"><hr class=\\"HairlineDivider__StyledHairlineDivider-biamlk-0 BvZCb\\"><p class=\\"Paragraph__StyledParagraph-efl81j-0 jukEfT\\">TELUS Easy Payment and Bring-It-Back applied. Read legal footnote Taxes due upfront.</p></div></div></div></div><div class=\\"Col__StyledCol-sc-15yvjc7-0 cFUOhb\\"><div class=\\"sc-bdvvtL iRVQej\\"><div class=\\"sc-bdvvtL bzuMLt sc-hKwDye klxlDh sc-hKwDye klxlDh\\"><div class=\\"sc-bdvvtL iEjIjW\\"><div class=\\"sc-bdvvtL eCDRPO\\"><img src=\\"../public/images/watch.jpg\\" width=\\"200\\" height=\\"300\\" alt=\\"Image of an IPhone\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"></div><div class=\\"sc-bdvvtL ipXlaM sc-pVTFL eoipii sc-pVTFL eoipii\\"><span data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 kwSJnX\\">Apple</span><span class=\\"sc-jrQzAO bBBXUK\\">Apple Watch SE</span></div><div class=\\"sc-bdvvtL iEjIjW sc-jRQBWg iapZSH sc-jRQBWg iapZSH\\"><div class=\\"sc-bdvvtL ipXlaM\\"><div class=\\"PriceLockup__StyledRateTextWrapper-crmhv4-4 inieGs\\"><div class=\\"sc-bdvvtL chLtDX sc-gKclnd gQsEUy sc-gKclnd gQsEUy\\"><div class=\\"sc-bdvvtL chLtDX\\"><span aria-hidden=\\"false\\" data-testid=\\"dollarSign\\" class=\\"PriceLockup__StyledDollarSign-crmhv4-2 bsmnqF\\">$</span><span data-testid=\\"priceValue\\" aria-hidden=\\"false\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\">369</span><span data-testid=\\"priceValue\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\"><span class=\\"Text__StyledText-sc-1m0rr67-0 sc-iCfMLu fMvSxW fqtOUC\\"></span></span></div></div></div></div></div></div><div class=\\"sc-bdvvtL iaciPl\\"><hr class=\\"HairlineDivider__StyledHairlineDivider-biamlk-0 BvZCb\\"><p class=\\"Paragraph__StyledParagraph-efl81j-0 jukEfT\\">TELUS Easy Payment and Bring-It-Back applied. Read legal footnote Taxes due upfront.</p></div></div></div></div><div class=\\"Col__StyledCol-sc-15yvjc7-0 cFUOhb\\"><div class=\\"sc-bdvvtL iRVQej\\"><div class=\\"sc-bdvvtL bzuMLt sc-hKwDye klxlDh sc-hKwDye klxlDh\\"><div class=\\"sc-bdvvtL iEjIjW\\"><div class=\\"sc-bdvvtL eCDRPO\\"><img src=\\"../public/images/watch.jpg\\" width=\\"200\\" height=\\"300\\" alt=\\"Image of an IPhone\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"></div><div class=\\"sc-bdvvtL ipXlaM sc-pVTFL eoipii sc-pVTFL eoipii\\"><span data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 kwSJnX\\">Apple</span><span class=\\"sc-jrQzAO bBBXUK\\">Apple Watch Series 3</span></div><div class=\\"sc-bdvvtL iEjIjW sc-jRQBWg iapZSH sc-jRQBWg iapZSH\\"><div class=\\"sc-bdvvtL ipXlaM\\"><div class=\\"PriceLockup__StyledRateTextWrapper-crmhv4-4 inieGs\\"><div class=\\"sc-bdvvtL chLtDX sc-gKclnd gQsEUy sc-gKclnd gQsEUy\\"><div class=\\"sc-bdvvtL chLtDX\\"><span aria-hidden=\\"false\\" data-testid=\\"dollarSign\\" class=\\"PriceLockup__StyledDollarSign-crmhv4-2 bsmnqF\\">$</span><span data-testid=\\"priceValue\\" aria-hidden=\\"false\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\">259</span><span data-testid=\\"priceValue\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\"><span class=\\"Text__StyledText-sc-1m0rr67-0 sc-iCfMLu fMvSxW fqtOUC\\"></span></span></div></div></div></div></div></div><div class=\\"sc-bdvvtL iaciPl\\"><hr class=\\"HairlineDivider__StyledHairlineDivider-biamlk-0 BvZCb\\"><p class=\\"Paragraph__StyledParagraph-efl81j-0 jukEfT\\">TELUS Easy Payment and Bring-It-Back applied. Read legal footnote Taxes due upfront.</p></div></div></div></div></div></div>"`;
